server/events/vcs/github_client.go

```diff
@@ -307,6 +309,40 @@ func (g *GithubClient) PullIsMergeable(repo models.Repo, pull models.PullRequest
 	return true, nil
 }
 
+// Check if the Pull Request is locked with :lock emoji
+func (g *GithubClient) PullIsLocked(repo models.Repo, pull models.PullRequest) (bool, error) {
+	statuses, err := g.getRepoStatuses(repo, pull)
+	if err != nil {
+		return false, errors.Wrapf(err, "fetching repo statuses for repo: %s, and pull number: %d", repo.FullName, pull.Num)
+	}
+
+	for _, status := range statuses {
+		if status.GetContext() == SubmitQueueReadinessStatusContext {
+			description := make(map[string]interface{})
+			err := json.Unmarshal([]byte(status.GetDescription()), &description)
```

Comment:
Usually in go you can use tags on struct fields to marshal and unmarshal json.
https://pkg.go.dev/encoding/json

I think it's probably better that you used this approach though since there is no pre-defined schema for submit queue and the json field can literally contain anything. I think it's worth writing that as a comment here for future readers of this code.

Reply:
Done!